moveit_core/online_signal_smoothing/src/ruckig_filter.cpp

```diff
@@ -0,0 +1,194 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2024, Andrew Zelenak
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <moveit/online_signal_smoothing/ruckig_filter.h>
+#include <rclcpp/clock.hpp>
+#include <rclcpp/logging.hpp>
+
+// Disable -Wold-style-cast because all _THROTTLE macros trigger this
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+
+namespace online_signal_smoothing
+{
+namespace
+{
+rclcpp::Logger getLogger()
+{
+  return moveit::getLogger("moveit.core.ruckig_filter_plugin");
+}
+inline constexpr double DEFAULT_JERK_BOUND = 300;  // rad/s^3
```

Comment:
i don't believe you can omit jerk limits in ruckig. Would be interested to see a link about that.

Reply:
![image](https://github.com/user-attachments/assets/d1ff95f9-96b8-4acf-8191-d32d19756633)

from https://docs.ruckig.com/index.html

Reply:
That's cool, probably a new feature. Made the jerk bounds `std::optional`

Reply:
I gave the `std::optional` jerk limits a college try but it crashed. So, the jerk limits are strictly required now.

Reply:
Give me a few days to experiment and I'll get back to you...

Reply:
Alright, I got it working with optional jerk limits  :+1:

Reply:
Yeah, I can't get optional jerk limits working. Made a Ruckig issue [here](https://github.com/pantor/ruckig/issues/200). It may be a bug on their side.

Reply:
Back to strictly requiring jerk limits, and opened the PR to `moveit_resources` to add jerk limits for the Panda, as you suggested. https://github.com/moveit/moveit_resources/pull/201